test-project/Assets/EditmodeTests/Subscriptions/InjectionCriteriaTests.cs

```diff
@@ -12,118 +13,161 @@ namespace Improbable.Gdk.EditmodeTests.Subscriptions
     ///     This tests the injection criteria for a reader or writer.
     /// </summary>
     [TestFixture]
-    public class ReaderWriterInjectionCriteriaTests : SubscriptionsTestBase
+    public class ReaderWriterInjectionCriteriaTests : MockBase
     {
         private const long EntityId = 100;
 
-        private GameObject createdGameObject;
-
-        [SetUp]
-        public override void Setup()
-        {
-            base.Setup();
-
-            var template = new EntityTemplate();
-            template.AddComponent(new Position.Snapshot(), "worker");
-            ConnectionHandler.CreateEntity(EntityId, template);
-            ReceiveSystem.Update();
-        }
-
-        [TearDown]
-        public override void TearDown()
-        {
-            base.TearDown();
-            Object.DestroyImmediate(createdGameObject);
-        }
-
         [Test]
         public void Reader_is_not_injected_when_entity_component_is_not_checked_out()
         {
-            createdGameObject = CreateAndLinkGameObjectWithComponent<ExhaustiveSingularReaderBehaviour>(EntityId);
-            var readerBehaviour = createdGameObject.GetComponent<ExhaustiveSingularReaderBehaviour>();
-
-            Assert.IsFalse(readerBehaviour.enabled);
-            Assert.IsNull(readerBehaviour.Reader);
+            World
+                .Step(world => { world.Connection.CreateEntity(EntityId, BasicEntity()); })
+                .Step(world =>
+                {
+                    var (_, readerBehaviour) = world.CreateGameObject<ExhaustiveSingularReaderBehaviour>(EntityId);
+                    return readerBehaviour;
+                })
+                .Step((world, reader) =>
```

Comment:
This does change the test? As there is now a step between creation and assertion.

Reply:
There was an `Update` in the old `CreateAndLinkGameObject<T>` - https://github.com/spatialos/gdk-for-unity/blob/master/test-project/Assets/TestBases/SubscriptionsTestBase.cs#L45